File: Plataformas/HackerRank/Closest_Numbers.js
let inputString = "";
const closestPairs = [];

function resetClosestPairsArray() {
	closestPairs.splice(0, closestPairs.length);
}

process.stdin.on("data", function (chunk) {
	inputString += chunk;
});

process.stdin.on("end", function () {
	main();
});

function main() {
	const lines = inputString.trim().split("\n");
	const n = parseInt(lines[0]);
	const arr = lines[1].split(" ").map(Number);
	arr.sort(function (a, b) { return a - b;});

	let smallest_diff = Math.abs(arr[0] - arr[1]);
    closestPairs.push(arr[0])
    closestPairs.push(arr[1])
    for (let i = 2; i < n; i++) {
        const diff = Math.abs(arr[i-1] - arr[i]);
        if (diff === smallest_diff) {
            closestPairs.push(arr[i-1])
            closestPairs.push(arr[i])
        }
        else if (diff < smallest_diff) {
            resetClosestPairsArray();
            smallest_diff = diff;
            closestPairs.push(arr[i-1])
            closestPairs.push(arr[i])        
        }
    }

    process.stdout.write(closestPairs.join(' ') + '\n');
}

File: Plataformas/HackerRank/Closest_Numbers.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    vector<pair<int, int>> answer;
    int smallest_diff = INT_MAX;
    int n;
    cin >> n;

    int arr[n];
    for (int i = 0; i < n; i++) cin >> arr[i];

    sort(arr, arr + n);

    smallest_diff = abs(arr[0] - arr[1]);
    answer.push_back({arr[0], arr[1]});
    for (int i = 2; i < n; i++) {
        int diff = abs(arr[i] - arr[i-1]);
        if (diff == smallest_diff)
            answer.push_back({arr[i-1], arr[i]});
        else if (diff < smallest_diff) {
            smallest_diff = diff;
            answer.clear();
            answer.push_back({arr[i-1], arr[i]});
        }
    }

    printf("%d %d", answer[0].first, answer[0].second);
    for (int i = 1; i < answer.size(); i++) {
        printf(" %d %d", answer[i].first, answer[i].second);
    }
    cout << "\n";
}